torch/random.py

```diff
@@ -29,17 +30,25 @@ def get_rng_state() -> torch.Tensor:
     return default_generator.get_state()
 
 
-def manual_seed(seed) -> torch._C.Generator:
-    r"""Sets the seed for generating random numbers on all devices. Returns a
-    `torch.Generator` object.
+def manual_seed(seed, device: Union[str, torch.device] = None) -> torch._C.Generator | None:
+    r"""Sets the seed for generating random numbers on all devices or on the
+    current device of a given device type. Returns a `torch.Generator` object.
 
     Args:
         seed (int): The desired seed. Value must be within the inclusive range
             `[-0x8000_0000_0000_0000, 0xffff_ffff_ffff_ffff]`. Otherwise, a RuntimeError
             is raised. Negative inputs are remapped to positive values with the formula
             `0xffff_ffff_ffff_ffff + seed`.
+        device (str): The device to set the seed.
     """
     seed = int(seed)
+
+    # Set the seed for the current device of a given device type
+    if device is not None:
+        device_type = device.type if isinstance(device, torch.device) else device
+        _manual_seed_for_device(seed, device_type)
+        return None
+
     import torch.cuda
 
     if not torch.cuda._is_in_bad_fork():
```

Comment:
Take CUDA as an example:
- If set device="cuda", `torch.cuda.manual_seed()` will be called, the seed will be set on the current GPU
- if not, `torch.cuda.manual_seed_all` will be called, the seed will be set on all GPUs